lxd/instance/drivers/driver_utils.go

```diff
@@ -44,3 +46,15 @@ func SetupSharedMounts() error {
 	sharedMounted = true
 	return nil
 }
+
+// PrepareEqualTest modifies any unexported variables required for reflect.DeepEqual to complete safely.
+// This is used for tests to avoid infinite recursion loops.
+func PrepareEqualTest(insts ...instance.Instance) {
```

Comment:
If only used by tests, can it be in _test.go or doesn't that work across imports?

Reply:
@stgraber I had considered moving this test into the `instance/drivers` package as would be more appropriate there. However it is tightly coupled to the `containerTestSuite` type which is embedding the `lxdTestSuite` both of which are defined in `main`. Furthermore the test itself uses `instanceCreateInternal` also only in main. So it was turning into a much larger change in an already large PR to fix. I initially tried just removing the lines that set the internal variables in `lxc` to nil and false, however this seems to cause issues with `reflect.DeepEqual` used under the hood by the testify package, and resulted in an infinite loop/recursion that caused my machine to lock up and needed hard resetting :)

So I need a way to modify the internal state of the `lxc` type without moving the whole test suite. Although its not ideal.

Reply:
Right, my question was whether exporting this from a driver_utils_test.go file would still have it be picked up when running tests in the main package. If not, then yeah, we don't really have a choice for now, but if it is, then it would save us from exporting that function.

Alternatively, we may be able to set those entries to `nil` by using reflect too.

Reply:
@stgraber OK yes I see, I will try exporting it from another test, if not I will try to use reflect to avoid the need for that function.